### PR TITLE
(CFACT-232) Move to more common setup in leatherman

### PR DESCRIPTION
**Pull request by branan**
* Common preprocessor stuff for boost
* Coveralls setup
* cpplint/cppcheck targets

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -109,36 +98,10 @@ endif()
...
-if (COVERALLS)
-    set(FACTER_CXX_FLAGS "${FACTER_CXX_FLAGS} -fprofile-arcs -ftest-coverage")
-endif()
-
 # Force all binaries to be created in the same location.
 set(EXECUTABLE_OUTPUT_PATH ${PROJECT_BINARY_DIR}/bin)
-if (WIN32)
-    # On Windows, DLL paths aren't hardcoded in the executable. We place all the executables and libraries
-    # in the same directory to avoid having to setup the DLL search path in the dev environment.
-    set(LIBRARY_OUTPUT_PATH    ${PROJECT_BINARY_DIR}/bin)
```
Comment (MikaelSmith) [inline CMakeLists.txt]: Still want this on Windows.

Comment (branan) [inline CMakeLists.txt]: whoops, block delete.

**Comment by branan** [comment]
@MikaelSmith OK, fixed up the output path for windows (and shuffled some things back around to minimize the diff)

**Comment by MikaelSmith** [comment]
:+1:

**Pull request merged**